File: backend/src/models/preferences.ts
import { z } from 'zod';

export const UserPreferencesSchema = z.object({
  theme_mode: z.enum(['light', 'dark', 'system']).default('system'),
  use_mobile_data: z.boolean().default(false),
  share_location: z.boolean().default(false),
});

export const UserPreferencesResponseSchema = UserPreferencesSchema.extend({
  firebase_uid: z.string(),
});

export type UserPreferences = z.infer<typeof UserPreferencesSchema>;
export type UserPreferencesResponse = z.infer<typeof UserPreferencesResponseSchema>;

File: models/preferences.py
"""Pydantic models for user preferences."""
from typing import Literal

from pydantic import BaseModel


class UserPreferences(BaseModel):
    """User preferences model matching the database schema."""

    theme_mode: Literal["light", "dark", "system"] = "system"
    use_mobile_data: bool = False
    share_location: bool = False


class UserPreferencesResponse(UserPreferences):
    """Response model with Firebase UID."""

    firebase_uid: str
